packages/opentelemetry-exporter-collector/src/platform/node/sendSpans.ts

```diff
@@ -0,0 +1,104 @@
+/*!
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const http = require('http');
+const https = require('https');
+
+import { IncomingMessage } from 'http';
+import { hrTime, hrTimeToTimeStamp } from '@opentelemetry/core';
+import { CollectorExporter } from '../../CollectorExporter';
+
+import * as collectorTypes from '../../types';
+
+const url = require('url');
+
+/**
+ * function that is called once when {@link ExporterCollector} is initialised
+ * in node version this is not used
+ * @param shutdownF shutdown method of {@link ExporterCollector}
+ */
+export function onInit(shutdownF: Function) {}
+
+/**
+ * function to be called once when {@link ExporterCollector} is shutdown
+ * in node version this is not used
+ * @param shutdownF - shutdown method of {@link ExporterCollector}
+ */
+export function onShutdown(shutdownF: Function) {}
+
+/**
+ * function to send spans to the [opentelemetry collector]{@link https://github.com/open-telemetry/opentelemetry-collector}
+ *     using the standard http/https node module
+ * @param spans
+ * @param onSuccess
+ * @param onError
+ * @param collectorExporter
+ */
+export function sendSpans(
+  spans: collectorTypes.Span[],
+  onSuccess: () => void,
+  onError: (status?: number) => void,
+  collectorExporter: CollectorExporter
+) {
+  const exportTraceServiceRequest: collectorTypes.ExportTraceServiceRequest = {
+    node: {
+      identifier: {
+        hostName: collectorExporter.hostName,
+        startTimestamp: hrTimeToTimeStamp(hrTime()),
+      },
+      libraryInfo: {
+        language: collectorTypes.LibraryInfoLanguage.NODE_JS,
+        // coreLibraryVersion: , not implemented
+        // exporterVersion: , not implemented
+        // coreLibraryVersion: , not implemented
+      },
+      serviceInfo: {
+        name: collectorExporter.serviceName,
+      },
+      // attributes: {}
+    },
+    // resource: '', not implemented
+    spans,
+  };
+  const body = JSON.stringify(exportTraceServiceRequest);
+  const parsedUrl = url.parse(collectorExporter.url);
+
+  const options = {
+    hostname: parsedUrl.hostname,
+    port: parsedUrl.port,
+    path: parsedUrl.path,
+    method: 'POST',
+    timeout: 5000,
+    headers: {
+      'Content-Length': Buffer.byteLength(body),
+    },
+  };
+
+  const request = parsedUrl.protocol === 'http:' ? http.request : https.request;
```

Comment:
AFAIK Collector receiver expects `gRPC` incoming requests, I don't think this will work. Did you get a chance to run and verify this?

Reply:
it is working fine
[tracing](https://ibb.co/JFcsthC)

Reply:
I testing this using sendBeacon, XMLHttpRequest and http from node. Am I missing something then ?

Reply:
Well, the collector can accept either HTTP or gRPC from either the browser or Node, so it will work fine from Node with HTTP. But gRPC is more performance/efficient from Node and we used it in the OpenCensus Node agent exporter ([code link](https://github.com/census-instrumentation/opencensus-node/blob/57e82f4a6ba3178c6e5797a2c1cb41140dd375bc/packages/opencensus-exporter-ocagent/src/ocagent.ts#L27))

But using gRPC here would likely be a larger change because the types may be slightly different for gRPC vs. HTTP and the most efficient way to do that would be to directly convert ReadableSpan into the gRPC types rather than to the grpc-gateway JSON and then to the gRPC types. I could be wrong on this, but this is what I expect.

What would you think about just creating a tracking issue for it so we can come back to it in the future if needed (it's really more of a performance improvement that something strictly needed)?

Reply:
this is good idea, maybe it could be then configurable via option, whether someone want to use http or gRpc ?

Reply:
So for the web we should always use HTTP, either `sendBeacon` where available or XHR. It's only for Node that it would be a realistic option to use gRPC (there is "gRPC Web", but that requires a lot more JS code, which isn't worth it to me).

For Node, I believe it's strictly better to use gRPC than HTTP, so I would suggest that would just be the standard way to do it once we implement it. I suppose it could be an option though as there might be occasional cases where gRPC wouldn't work due to some network requirements or something, but that feels unlikely in the Node case.

Reply:
https://github.com/open-telemetry/opentelemetry-js/issues/567